sharding/data_modeling.rst

```diff
@@ -36,276 +45,40 @@ The following diagram illustrates co-location in the multi-tenant data model. It
-
-    comment TEXT,
-    minor BOOLEAN,
-    type TEXT,
-
-    old_length INT,
-    new_length INT
-  );
-
-  SELECT create_distributed_table('wikipedia_editors', 'editor');
-  SELECT create_distributed_table('wikipedia_changes', 'editor');
-
-These tables can be populated by the Wikipedia API, and we can distribute them in Citus by the :code:`editor` column. Notice that this is a text column. Citus' hash distribution uses PostgreSQL hashing which supports a number of data types.
+Real-time queries typically ask for numeric aggregates grouped by date or category. Citus sends these queries to each shard for partial results and assembles the final answer on the coordinator node. Queries run fastest when as many nodes contribute as possible, and when no individual node bottlenecks.
 
-A co-located JOIN between editors and changes allows aggregates not only by editor, but by properties of an editor. For instance we can count the difference between the number of newly created pages by bot vs human. The grouping and counting is performed on worker nodes in parallel and the final results are merged on the coordinator node.
+Best Practices
+^^^^^^^^^^^^^^
 
-.. code-block:: postgres
+* **Choose a column with high cardinality as the distribution column.**
+  For comparison, a "status" field on an order table with values "new," "paid," and "shipped" is a poor choice of distribution column because it assumes only those few values. The number of distinct values limits the number of shards that can hold the data, and the number of nodes that can process it. Among columns with high cardinality, it is good additionally to choose those that are frequently used in group-by clauses or as join keys.
 
-  SELECT bot, count(*) AS pages_created
-  FROM wikipedia_changes c,
-       wikipedia_editors e
-  WHERE c.editor = e.editor
-    AND type = 'new'
-  GROUP BY bot;
+* **Choose a column with even distribution.**
+  If you distribute a table on a column skewed to certain common values, then data in the table will tend to accumulate in certain shards. The nodes holding those shards will end up doing more work than other nodes.
 
-Events and Reference Tables
-~~~~~~~~~~~~~~~~~~~~~~~~~~~
+* **Distribute fact and dimension tables on their common columns.**
+  Your fact table can have only one distribution key. Tables that join on another key will not be co-located with the fact table. Choose one dimension to co-locate based on how frequently it is joined and the size of the joining rows. 
 
-We've already seen how every row in a distributed table is stored on a shard. However for small tables there is a trick to achieve a kind of universal :ref:`co-location <colocation>`. We can choose to place all its rows into a single shard but replicate that shard to every worker node. It introduces storage and update costs of course, but this can be more than counterbalanced by the performance gains of read queries.
+* **Change some dimension tables into reference tables.**
```

Comment:
Link to reference table section here? As I noted above, maybe a brief paragraph earlier on which talks about reference tables might be useful.

Reply:
Link added.